kernel_module/uio/uio_ivshmem.c

```diff
@@ -101,12 +111,27 @@ static int request_msix_vectors(struct ivshmem_info *ivs_info, int nvectors)
 
 	if (err)
 		goto error;
+#else
+	err = pci_alloc_irq_vectors(ivs_info->dev, 1, ivs_info->nvectors,
+			PCI_IRQ_MSIX);
+
+	if (err <= 0) {
+		dev_info(&ivs_info->dev->dev,
+			 "no MSI (%d). Back to INTx.\n", err);
+		goto error;
+	} else {
+		ivs_info->nvectors = err;
+	}
+#endif
```

Comment:
I think with a bit of restructuring both the error handling and the assignment of "nvectors = err" can be reused across the two cases.